### PR TITLE
Myst rc 1

### PR DESCRIPTION
**Pull request by MystLeissa**
## About The Pull Request

This is mostly an personal content update - I'm offering it to the master branch for review  in case they actually want to incorporate it into the servers core preformance. This pull request is more of a formality because I'm pretty sure it's going to be rejected anyway nor do I expect it to be accepted - but if it is that's great! I tried to use the in-pull change log but a list of changes can be found in the fork under "changelogfork.txt" file

## Why It's Good For The Game

Additional Content is never bad, also these are some changes that have been kinda messing with my head as to why they aren't in the game yet. My main reason for issuing a pull request is because these are things that will never see the light of a "real" server in the event nobody incorporates it - since I'm pretty much playing on my own with this fork. Some of the changes are controversial i'll admit but I'm open to feedback.

## Changelog
:cl:
add: Lockable School Girl Uniforms
add: Miner Firing Pins
add: Miner Implants
add: Sleep Sack (pink bodybag kinda)
add: Girly Vendor (for silly girly things)
add: girlkey (a kind of specialized key to the uniforms with some extra effects thrown in)
add: girlkey lockbox to protect girlkeys
add: Holopad in Lavaland Public Camp
add: Weapons recharger in Miner's Camp
add: Disposal Chute to Lavaland Bar
add: Cargo Packs now include lockable schoolgirl crate
add: Sleep Sack to one of the BoxStation Dorms
add: Girly Vendor to BoxStation AI Antechamber
imageadd: pink schoolgirl uniform
imageadd: Girly Vendor
imageadd: Girly Resupply
imagedel: Old Body bag
imageadd: New Body Bag
imageadd: Sleep Sack
imageadd: Girly Crate
refactor: Moved Schoolgirl outfits to dedicated code file
balance: Most Firing Pins don't spawn with the gun anymore
balance: Kinetic Accelerators Spawn with Miner's Pins
balance: "Round Start" Miners spawn with a miner implant
tweak: Mecha can no longer be driven with the "DUMB" trait
code: Holodeck responds to girlkey by deactivating the derez check (but not adding the programs from disabling safeties)
code:Mob Trait "NOSUIT" - Causes spacesuits/suits (outerwear) to drop on wear
server: Lockable Schoolgirl uniforms aren't ALL bad effects - they do come with 4 "extra" secret pockets
server: Only Captain Access Can unlock girlkey lockboxes
server: Because of the firing pin changes you can expect the violence factor with guns to go WAY down
server: The Aforementioned also has the perk of giving security a reason to request firing pins from the warden
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
gay

**Comment by actioninja** [comment]
It doesn't, it's just not mapmerged at all so the diff is fucked up.

**Comment by Rowelll** [comment]
i think it (or at least a part) would really get merged if changes were a bit separated, now it just can be a matter of praying

**Comment by MystLeissa** [comment]
> gay
> edit: nevermind it removes lavaland and boxstation so this is based and redpilled

what do you mean? I had no end to the trouble getting this on Github - and your telling me it won't run?

**Comment by MystLeissa** [comment]
> It doesn't, it's just not mapmerged at all so the diff is fucked up.

I'm new at this github PR thing so would you mind explaining what mapmerging entails?

**Comment by MystLeissa** [comment]
> i think it (or at least a part) would really get merged if changes were a bit separated, now it just can be a matter of praying

I can try to seperate it - but it's just I did this all on my own - like on my PC - before even setting up the repository so it's kinda working with unideal conditions...

**Comment by Akrilla** [comment]
I think you can merge some of that changelog together jesus

**Comment by mrhugo13** [comment]
> 
> 
> > i think it (or at least a part) would really get merged if changes were a bit separated, now it just can be a matter of praying
> 
> I can try to seperate it - but it's just I did this all on my own - like on my PC - before even setting up the repository so it's kinda working with unideal conditions...

https://tgstation13.org/wiki/Map_Merger

File: code/game/objects/items/sleepsack.dm [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/obj/item/sleepsack
```
Comment (nemvar) [inline code/game/objects/items/sleepsack.dm]: Is this literally just a copy pasta of closets/bodybags?

**Review by Tlaltecuhtli — APPROVED** [review]
good for me merge this

File: changelogfork.txt [hunk 2]
```diff
@@ -0,0 +1,43 @@
+-Segments schoolgirl objects off and Creates Lockable Schoolgirl outfits
+   (Notes: All Schoolgirl/Anime Items were moved from Miscellanious to a new file called "schoolgirl.dm" in the same directory)
+-All Lockable Schoolgirl Outfits Share these charateristics:
```
Comment (Tlaltecuhtli) [inline changelogfork.txt]: wait a second

**Comment by Arianya** [comment]
Not to be cruel, I'm not questioning your intentions, but

>-All Lockable Schoolgirl Outfits Share these charateristics: 
	-adds DUMB Trait
	-contains NODROP trait
	-adds NOSUIT trait
	-unlocks with "girlkey"
	-unequips armor/outerwear slot when worn
	-Contains 4 "Hidden"/"Stealth" Pockets in addition to those on the normal HUD (alt-click on worn uniform)


Is this a weird fetish thing? It feels like a weird fetish thing.

File: code/modules/clothing/under/schoolgirl.dm [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+/obj/item/clothing/under/schoolgirl/locked
+	name = "pink schoolgirl uniform"
+	desc = "It has a little lock inside. It's also super cute."
+	icon_state = "schoolgirlpink"
+	item_state = "schoolgirlpink"
+	item_color = "schoolgirlpink"
+	has_sensor = NO_SENSORS
+	sensor_mode = SENSOR_COORDS
+	random_sensor = FALSE
```
Comment (nemvar) [inline code/modules/clothing/under/schoolgirl.dm]: Why even bother with those two vars if it doesn't have sensors?

**Comment by ninjanomnom** [comment]
You need to break up your changes into multiple prs for us to merge this. If you don't intend on actually working on those changes to get them merged into tg then you don't have to bother, I expect people will take code they want from this pr even when closed.

File: code/modules/clothing/under/schoolgirl.dm [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+/obj/item/clothing/under/schoolgirl/locked/dropped(mob/user)
+	. = ..()
+	wearer = null
+
+/obj/item/clothing/under/schoolgirl/locked/proc/toggle_lock(user)
+	locked = !locked
+	if(locked)
+		lock(user)
+	if(!locked)
+		unlock(user)
```
Comment (nemvar) [inline code/modules/clothing/under/schoolgirl.dm]: This entire proc is bad. You are already switching the locked var in the lock/unlock procs which makes a lot more sense. Why do you also do it here?

**Review by Fikou — COMMENTED** [review]
based, please open and merge i need to put slavelocks on my metaboyfriend eggar poop

**Comment by Akrilla** [comment]
I want my bimbo clothing on my cat boys NOW

**Comment by optimumtact** [comment]
rich with irony indeed

File: code/modules/projectiles/pins.dm [hunk 5]
```diff
@@ -114,7 +114,10 @@
 	icon_state = "firing_pin_pindi"
 	req_implant = /obj/item/implant/weapons_auth
 
-
+/obj/item/firing_pin/implant/miner
+	name = "Miner's Firing Pin"
+	desc = "Secures your miner's weapons"
```
Comment (ATH1909) [inline code/modules/projectiles/pins.dm]: *miners'

**Comment by ATH1909** [comment]
God-tier PR. Speedmerge.

**Comment by Krysonism** [comment]
OwO

**Comment by ATH1909** [comment]
Nemvar pointed out on Discord that this PR makes even magical staves and wands spawn without their special "firing pins", which is epic.

File: code/modules/clothing/under/schoolgirl.dm (hunk 4 above)
Comment (MystLeissa) [reply code/modules/clothing/under/schoolgirl.dm]: I was trying to clean up my code - originally I did the ENTIRE operation in that segmen - i moved it to allow for easier sorting of effects.

**Comment by DominikPanic** [comment]
good PR will port

**Comment by AnthonyJNgo** [comment]
We need to become a tg downstream again ASAP so we can merge this quality content

**Comment by gardenBanter** [comment]
lmaing @ur life

File: code/modules/projectiles/pins.dm [hunk 6]
```diff
@@ -150,9 +153,32 @@
...
+	desc = "Will only let females fire the weapon"
+	force_replace = 1
+	custom_price = 100
+	fail_message = "<span class='warning'>NOT A GIRL! FIRING REJECTED!!</span>"
+
+/obj/item/firing_pin/girl/gun_insert(mob/living/user, obj/item/gun/G)
+	var/gender = user.gender
+	var/user_name = user.real_name
+	var/msg = "[user_name] Is [gender]"
+	to_chat(GLOB.admins, msg)
```
Comment (Kmc2000) [inline code/modules/projectiles/pins.dm]: Nice

File: code/modules/vending/_vending.dm [hunk 7]
```diff
@@ -77,7 +77,7 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	var/extra_price = 50
 	var/onstation = TRUE //if it doesn't originate from off-station during mapload, everything is free
 	var/list/canload_access_list
-
+	var/girl_locked = FALSE
```
Comment (Kmc2000) [inline code/modules/vending/_vending.dm]: hmmm

File: changelogfork.txt [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+-Changed _suit, and _spacesuit code to drop-on-wear when aforementioned trait is present
+-Changed Mecha Code - No More Dumb Pilots (hopefully)
+-SLIGHTLY changes the body bag icon files (including existing body bag reicon - accidental feel free to overwrite the changed icon if you like it better)
+-Changes Cargo packs to include "Schoolgirl" Costumes
+-Changes Box Station Map to include
+	1. A Third Supply Request Terminal in the bar (for easy access)
+	2. Girly Stuff Vendor in Satellight Antechmber
+	3. Removes a few non-essential props (added back by myst-rc-1)
+	4. Adds a "Sleep sack" (basically a body bag) to one of the dorms
+		(note: Code Needs reviewing I really need to place it in the closest subdatum instead of relying on a completely original item file with mostly copied contents)
```
Comment (Kmc2000) [inline changelogfork.txt]: A subdatum you say?

**Review by Kmc2000 — COMMENTED** [review]
I don't know what to think

**Comment by Vivalas** [comment]
this is peak contribution

**Comment by Detective-Google** [comment]
on the bright side this makes for _great_ entertainment, my sides are in orbit.

_oh god oh fuck is this bait? this is so autistic I think we may have been jebaited, please tell me this isnt unironic_

**Comment by Vivalas** [comment]
>these are some changes that have been kinda messing with my head as to why they aren't in the game yet.

No I.. I don't think this is ironic. And neither are my comments. I have been persuaded. I have seen the light. Praise be to this PR.

**Comment by Ludox235** [comment]
Why the **FUCK** is this closed? This is a based PR, you should at the very least merge it for respect of the author to **somehow have a litle ounce of hope** that it gets merged.

Kids nowadays....

**Comment by optimumtact** [comment]
@MystLeissa your contributions are mostly out of tone with where /tg/ was aiming to head, which is why you got the reception got.

I recognise this can be a pretty depressing fact that an upstream isn't interested in the same things you are, but I believe you are within you rights to at least try and the behaviour shown by some was unnecessarily hostile

For future reference I doubt any of the changes you intended to make would find a home here.

However we always welcome systems and coding improvements upstream.

**Pull request closed without merging**